packages/expo-updates/ios/EXUpdates/EXUpdatesFileDownloader.m

```diff
@@ -0,0 +1,116 @@
+
+NSString * const kEXUpdatesFileDownloaderErrorDomain = @"EXUpdatesFileDownloader";
+NSTimeInterval const kEXUpdatesDefaultTimeoutInterval = 60;
+
+@interface EXUpdatesFileDownloader () <NSURLSessionDataDelegate>
+@end
+
+@implementation EXUpdatesFileDownloader
+
+- (void)downloadFileFromURL:(NSURL *)url
+                     toPath:(NSString *)destinationPath
+               successBlock:(EXUpdatesFileDownloaderSuccessBlock)successBlock
+                 errorBlock:(EXUpdatesFileDownloaderErrorBlock)errorBlock
+{
+  [self downloadDataFromURL:url successBlock:^(NSData * _Nonnull data, NSURLResponse * _Nonnull response) {
+    if ([[NSFileManager defaultManager] createFileAtPath:destinationPath contents:data attributes:nil]) {
+      successBlock(data, response);
+    } else {
+      errorBlock([NSError errorWithDomain:kEXUpdatesFileDownloaderErrorDomain code:-1 userInfo:@{NSLocalizedDescriptionKey: [NSString stringWithFormat:@"Could not write to path: %@", destinationPath]}], response);
+    }
+  } errorBlock:errorBlock];
+}
+
+- (void)downloadDataFromURL:(NSURL *)url
+               successBlock:(EXUpdatesFileDownloaderSuccessBlock)successBlock
+                 errorBlock:(EXUpdatesFileDownloaderErrorBlock)errorBlock
+{
+  NSURLSessionConfiguration *configuration = _urlSessionConfiguration ?: [NSURLSessionConfiguration defaultSessionConfiguration];
+
+  // also pass any custom cache policy onto this specific request
+  NSURLRequestCachePolicy cachePolicy = _urlSessionConfiguration ? _urlSessionConfiguration.requestCachePolicy : NSURLRequestUseProtocolCachePolicy;
+
+  NSURLSession *session = [NSURLSession sessionWithConfiguration:configuration delegate:self delegateQueue:nil];
+  NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:url cachePolicy:cachePolicy timeoutInterval:kEXUpdatesDefaultTimeoutInterval];
+  [self _setHTTPHeaderFields:request];
+
+  __weak typeof(self) weakSelf = self;
+  NSURLSessionDataTask *task = [session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+    if (!error && [response isKindOfClass:[NSHTTPURLResponse class]]) {
+      NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
+      if (httpResponse.statusCode != 200) {
+        NSStringEncoding encoding = [weakSelf _encodingFromResponse:response];
+        NSString *body = [[NSString alloc] initWithData:data encoding:encoding];
+        error = [weakSelf _errorFromResponse:httpResponse body:body];
+      }
+    }
+
+    if (error) {
+      errorBlock(error, response);
+    } else {
+      successBlock(data, response);
+    }
+  }];
+  [task resume];
+  [session finishTasksAndInvalidate];
+}
+
+- (void)_setHTTPHeaderFields:(NSMutableURLRequest *)request
+{
+  // TODO(eric): add more fields here
+  [request setValue:@"ios" forHTTPHeaderField:@"Expo-Platform"];
+
+  NSString *binaryVersion = [[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleShortVersionString"];
```

Comment:
Use modern Obj-C syntax (`.` for property access and `[]` for dictionary indexing):
```objc
NSString *binaryVersion = NSBundle.mainBundle.infoDictionary[@"CFBundleShortVersionString"];
```